src/cargo/core/compiler/build_runner/compilation_files.rs

```diff
@@ -25,6 +25,21 @@ use crate::util::{self, CargoResult, StableHasher};
 /// use the same rustc version.
 const METADATA_VERSION: u8 = 2;
 
+#[derive(Copy, Clone, Hash, Eq, PartialEq, Ord, PartialOrd)]
```

Comment:
Since this is exposed to other functions as returning type, this may worth a doc comment. Something like this:

```suggestion
/// A hash value uniquely identifying a [`Unit`] in a build. See [`Metadata`] for more.
#[derive(Copy, Clone, Hash, Eq, PartialEq, Ord, PartialOrd)]
```